src/writer/blocks/writeraskkg.py

```
@@ -93,31 +104,54 @@ def run(self):
+
+                        if graph_citations:
+                            delta_sources = chunk.model_extra.get("sources", "")
+                            citations_so_far.extend(delta_sources)
+                            result_dict["citations"] = citations_so_far
+                        
+                        self._set_state(state_element, result_dict)
+
                     except json.JSONDecodeError:
-                        logging.error(
+                        logger.error(
                             "Could not parse stream chunk from graph.question")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)